src/services/userService.ts

```diff
@@ -1,4 +1,13 @@
 import api from 'config/api';
 import { ApiError } from 'utils/types';
 
+import LocalStorageService from './LocalStorageService';
+
 export const signUp = <T>(data: T) => api.post<Response, ApiError>('/users', data);
+
+export const login = <T>(data: T) =>
+  api.post<Response, Error>('/users/sign_in', data).then(response => {
+    const token = response.headers?.['access-token'];
+    LocalStorageService.setValue('session', token);
```

Comment:
this should not be done here, its better if you return the response as `{...response.data, token: response.headers?.['access-token']}` than placing logic like localstorage service here. It is more explicit if you keep using `onSuccess` for behaviours that does not anything to do with the service itself